packages/assets-controllers/src/TokenDetectionController.ts

```diff
@@ -285,11 +405,31 @@ export class TokenDetectionController extends StaticIntervalPollingController<
   }
 
   /**
-   * Triggers asset ERC20 token auto detection for each contract address in contract metadata on mainnet.
+   * Restart token detection polling period and call detectNewTokens
+   * in case of address change or user session initialization.
    *
-   * @param options - Options to detect tokens.
+   * @param options - Options for restart token detection.
+   * @param options.selectedAddress - the selectedAddress against which to detect for token balances
    * @param options.networkClientId - The ID of the network client to use.
-   * @param options.accountAddress - The account address to use.
+   */
+  async #restartTokenDetection({
```

Comment:
Revisiting this again, I really wonder if we should have this method at all. This doesn't "restart" anything, it just refetches tokens and then resets the interval for some reason. What if we have all of the event listeners in the constructor call `detectTokens` directly?

Reply:
We decided to keep this behavior, since this is what the extension does.